File: mobile/src/main/java/treehou/se/habit/ui/colorpicker/LightModule.java
package treehou.se.habit.ui.colorpicker;

import android.os.Bundle;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;
import treehou.se.habit.module.ViewModule;

@Module
public class LightModule extends ViewModule<LightFragment> {

    protected final Bundle args;

    public LightModule(LightFragment fragment, Bundle args) {
        super(fragment);
        this.args = args;
    }


    @Provides
    public LightContract.View provideView() {
        return getView();
    }

    @Provides
    public LightContract.Presenter providePresenter(LightPresenter presenter) {
        return presenter;
    }

    @Provides
    @Named("arguments")
    public Bundle provideArgs() {
        return args;
    }
}

File: mobile/src/main/java/treehou/se/habit/ui/colorpicker/LightModule.kt
package treehou.se.habit.ui.colorpicker

import android.os.Bundle
import dagger.Module
import dagger.Provides
import treehou.se.habit.module.ViewModule
import javax.inject.Named

@Module
class LightModule(fragment: LightFragment, protected val args: Bundle) : ViewModule<LightFragment>(fragment) {

    @Provides
    fun provideView(): LightContract.View? {
        return view
    }

    @Provides
    fun providePresenter(presenter: LightPresenter): LightContract.Presenter {
        return presenter
    }

    @Provides
    @Named("arguments")
    fun provideArgs(): Bundle {
        return args
    }
}
